keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -471,13 +455,22 @@ status:
   - type: Ready
     status: "True"
     lastTransitionTime: "2020-03-30T01:33:51Z"
-  - type: InvalidService
+    reason: "Ready"
```

Comment:
nit but I don't think this should merely duplicate the type - if there's no other short reason that adds value then leave it blank.

Reply:
I now remember why I did this, we actually cannot add a Condition with an empty reason

Reply:
Do you mean the API service rejects it?

Reply:
I am not sure who rejects it but I retested it and by removing one reason I got the following error:
```
time="2024-09-26T10:12:11.051078297Z" level=error msg="Reconciler error" ServiceImport="{rebel-base-mcsapi default}" controller=serviceimport controllerGroup=multicluster.x-k8s.io controllerKind=ServiceImport error="ServiceExport.multicluster.x-k8s.io \"rebel-base-mcsapi\" is invalid: status.conditions[0].reason: Invalid value: \"\": conditions[0].reason in body should be at least 1 chars long" name=rebel-base-mcsapi namespace=default reconcileID="\"9d39229b-9e9b-4c8d-81d8-5a6cfda7dd68\"" subsys=controller-runtime

```

Reply:
K8s rejects it b/c `Reason` is required in `metav1.Condition`:

```
// This field may not be empty
// +required
// +kubebuilder:validation:Required
// +kubebuilder:validation:MaxLength=1024
// +kubebuilder:validation:MinLength=1
// +kubebuilder:validation:Pattern=`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`
Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
```

In the previous [ServiceExportCondition](https://github.com/kubernetes-sigs/mcs-api/blob/d26176718973179fb643f9e7408457ee3430139e/pkg/apis/v1alpha1/serviceexport.go#L82) it wasn't.